### DISCUSSION TITLE
It is not possible to create RPM source package with additional macros

### DISCUSSION THREAD

**Discussion by uddr** [rpm-software-management/mock#1069] [General]
### Short description of the problem
It is not possible to create RPM source package with additional macros. 'buildsrpm' command does not include 'rpmbuild_opts'. The source packages which are built by `rpmbuild` and `mock` are different.
With `rebuild` command this is [possible](https://github.com/rpm-software-management/mock/blob/cab12758da074e8af87189bd8f4a11c73a1f1ebf/mock/py/mockbuild/backend.py#L735)

### Output of `rpm -q mock`

mock-3.5-1.el8.noarch

### Steps to reproduce issue

1. Build RPM source package via `rpmbuild`

`rpmbuild -D '_buildhost docker' -D 'source_date_epoch_from_changelog 1' -D '_topdir /root/rpmbuild' -bs file.spec 

2. Build RPM source package via `mock`

`mock -v --resultdir /root/rpmbuild/SRPMS --rpmbuild-opts="-D '_buildhost docker' -D 'source_date_epoch_from_changelog 1'  -D '_topdir /root/rpmbuild'" --buildsrpm --spec /root/rpmbuild/SPECS/openvpn3.spec --sources /root/rpmbuild/SOURCES

3. Compare those two packages (`diffoscope` is the best for this in my opinion) - they are different, additional macros are not applied.

Small diff from `diffoscope` output
```
│ -BUILDTIME: 1682996111
│ -BUILDHOST: 7cdd84d13d27
│ +BUILDTIME: 1682996698
│ +BUILDHOST: 734ffc1c6198
```

Do not forget to mention full commandline with the mock command you executed.

### Any additional notes

<details>
  <summary>
    <p>Output of `mock --debug-config`</p>
  </summary>

  PUT THE OUTPUT HERE
</details>

**Comment by xsuchy** [comment]
1) more convenient is `config_opts['macros']['%Add_your_macro_name_here'] = "add macro value here"`

2) especially for host you can use `config_opts['macros']['%_buildhost'] = 'my.own.hostname' `

**Comment by uddr** [comment]
Thanks for the suggestion. 
For `_buildhost` it should work but it seems macros like `source_date_epoch_from_changelog` is not possible to pass at all
https://github.com/rpm-software-management/mock/blob/cab12758da074e8af87189bd8f4a11c73a1f1ebf/mock/py/mockbuild/backend.py#L695

or did I miss something?

**Comment by xsuchy** [comment]
The macros specified in `config_opts['macros']` are not passed on command line, but are put in `~/.rpmmacros` that is read by rpmbuild.

**Reply by uddr** [reply]
Thanks for the response. I understand the idea. But let's take a look on this wider.
We have `buildsrpm` parameter which is actually `rpmbuild -bs`. And `rebuild` parameter which is `rpmbuild -bb` (not only but probably most common)

`rpmbuild` accepts macros (`-D`) for both (and they works).
But `mock` don't have `rpmbuild_opts` parameter for `buildsrpm` case. At all.
